agent/rpc_from.go

```diff
@@ -38,8 +28,41 @@ func (a *orbAgent) handleGroupMembership(rpc fleet.GroupMembershipRPCPayload) {
 	}
 }
 
-func (a *orbAgent) handleAgentPolicies(rpc []fleet.AgentPolicyRPCPayload) {
-
+func (a *orbAgent) handleAgentPolicies(rpc []fleet.AgentPolicyRPCPayload, fullList bool) {
+	if fullList {
+		policies, err := a.policyManager.GetRepo().GetAll()
+		if err != nil {
+			a.logger.Error("failed to retrieve policies on handle subscriptions")
+		}
+		// Compare the list of policies the check which one should be removed
+		// After a editing a policy
+		policyRemove := map[string]bool{}
+		for _, p := range policies {
+			policyRemove[p.ID] = true
+			for _, payload := range rpc {
+				if p.ID == payload.ID {
+					policyRemove[p.ID] = false
+				}
+			}
+		}
+		// Remove only the policy which should be removed
+		for k, v := range policyRemove {
+			policy, err := a.policyManager.GetRepo().Get(k)
+			if err != nil {
+				a.logger.Error("failed to retrieve policy", zap.String("policy_id", k), zap.Error(err))
+			}
+			if v == true {
+				a.policyManager.ManagePolicy(fleet.AgentPolicyRPCPayload{
```

Comment:
hm I don't like using ManagePolicy here with a mocked AgentPolicyRPCPayload. Let's refactor the logic out of ManagePolicy for "remove" to a common method we can call.